subprojects/platform-native/src/docs/guide-templates/building-native-application/README.adoc

```diff
@@ -0,0 +1,172 @@
+
+== What we’ll need
+
+* A text editor or IDE - for example https://www.jetbrains.com/idea/download/[IntelliJ IDEA]
+* A Java Development Kit (JDK), version 8 or higher - for example https://adoptopenjdk.net/[AdoptOpenJDK]
+* The latest https://gradle.org/install[Gradle distribution]
+* An installed C compiler.
+ See which link:TODO[C toolchains] are supported by Nokee.
+
+== 1. Create a project folder
+
+First, let's create an empty folder for our project and use the build-in `wrapper` task to create a Gradle wrapper script, e.g. `gradlew`.
+
+[source,terminal]
+----
+$ mkdir demo
+$ cd demo
+$ gradle wrapper
+
+BUILD SUCCESSFUL
+1 actionable tasks: 1 executed
+$ tree .
+├── gradle       // <1>
+│   └── wrapper
+│       ├── gradle-wrapper.jar
+│       └── gradle-wrapper.properties
+├── gradlew      // <2>
+└── gradlew.bat  // <2>
+
+2 directories, 4 files
+$ touch settings.gradle // <3>
+
+----
+
+<1> Generated folder for wrapper files
+
+<2> Gradle wrapper start scripts
+
+<3> Create an empty Gradle workspace buildscript
+
+We now have a Gradle project setup.
+Let's configure {C application} capability to the project.
+
+== 2. Configure {C application} capability
+
+First, create the project buildscript, e.g. `build.gradle`.
+
+build.gradle
+
+[source,groovy]
+----
+plugins {  // <1>
+	id 'dev.nokee.c-application' // <2>
+}
+----
+
+<1> We use the plugin block to apply plugins that extends the project capability.
+We can apply any Gradle project plugin here.
+Learn more about the link:TODO[plugins block in Gradle's documentation].
+Refer to each plugin's reference chapter to learn more about each individual plugins.
+
+<2> We apply Nokee's {C application} plugin.
+Learn more about link:TODO[this plugin at its reference chapter].
+
+NOTE: Gradle support two domain specific language (DSL): Groovy DSL and Kotlin DSL.
+For this guide, we will use Groovy DSL which is closer in syntax to {C} language.
+Learn more about Gradle DSL language in the link:TODO[Getting started chapter].
+
+== 3. Add {C} code
+
+By default, Nokee uses link:TODO[Gradle's conventional source layout].
+All {C} application code is located at `src/main/c` and its private headers are located at `src/main/headers`.
+
+Let's add a very simple hello world:
+
+.src/main/c/main.c
+[source,c]
+----
+#include <stdio>
+#include <my_string.h>
+
+int main(int argc, char** argv) {
+	printf("%s\n", kHelloWorld);
+	return 0;
+}
+----
+
+.src/main/headers/my_string.h
+[source,c]
+----
+static const char* kHelloWorld = "Hello, World!";
+----
+
+Next, let's build the application.
+
+== 3. Build the {application}
+
+[source,terminal]
+----
+$ ./gradlew build
+
+BUILD SUCCESSFUL
+7 actionable tasks: 7 executed
+----
+
+NOTE: The first time you run the wrapper script, e.g. `gradlew[.bat]`, there may be a delay while that version of gradle is downloaded and stored locally in your `~/.gradle/wrapper/dists` folder.
+Learn link:TODO[why Gradle download distribution when using wrapper].
```

Comment:
🔲 For users unfamiliar with Gradle, the described behaviour here may be surprising and hinder their understanding of Gradle. It's important to mention a quick call out to the first-time use of the wrapper script. I also think it would be useful to the user to have a dedicated page/section on why Gradle downloaded the distribution given that we already told them to install Gradle.

❓ Given that we already ask the user to install Gradle locally, should we disregard the wrapper script and focus only on the sample. In theory, users would use some sort of `init` task that will initialize a Gradle project with the correct wrapper. Users should always use a wrapper script. However, in practice, we don't have the `init` task and we are left with bootstrap paradoxes where you need Gradle to generate the wrapper and using the wrapper download the distribution which you already downloaded. I almost feel like it's more confusing to a new user.

👉 We should probably have a call out in the Next step section to guide explaining the Gradle wrapper and everything around it (including the reason why the wrapper downloads a distribution). We should also include that note in ready-made archive that includes all the code of the guide.